src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs

```diff
@@ -330,10 +355,17 @@ public override MethodImplAttributes GetMethodImplementationFlags()
         #endregion
 
         #region Invocation Logic(On MemberBase)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void CheckConsistency(object? target)
         {
             // only test instance methods
-            if ((m_methodAttributes & MethodAttributes.Static) != MethodAttributes.Static)
+            if ((m_methodAttributes & MethodAttributes.Static) == 0)
+            {
+                CheckConsistencyInstance(target);
+            }
+
+            [MethodImpl(MethodImplOptions.NoInlining)] // move check out of hot path when invoking static methods
```

Comment:
This is not a lazy initialization. Is splitting this method really improvement? Maybe mark the whole method with `AggressiveInlining` without spliting it?

Reply:
Splitting did end up providing a ~3% or so perf improvement which was consistent across several runs. But I didn't try inlining the whole thing. Will give that a shot and see how it goes.